packages/core/src/core/fetch.ts

```diff
@@ -186,21 +175,33 @@ const addFetchChangeMetadata = (
   ),
 }])
 
+const getChangesNestedUnderID = (
+  id: ElemID, changesTree: collections.treeMap.TreeMap<WorkspaceDetailedChange>
+): WorkspaceDetailedChange[] => (
+  wu(changesTree.valuesWithPrefix(id.getFullName()))
+    .flatten(true)
+    // Instance IDs are nested under the type ID in the tree, so we have to filter this
+    .filter(item => id.isEqual(item.change.id) || id.isParentOf(item.change.id))
```

Comment:
I know this is existing code but this seems pretty inefficient, and it seems like we only have a few edge cases left after `valuesWithPrefix`. 
maybe we can do something like
```suggestion
    .filter(item => id.createTopLevelParentID().parent === item.change.id.createTopLevelParentID().parent)
```
or something like that (we could of course compute the lhs only once) to make sure we're not including instances under types? is there any other scenario that we care about?

Reply:
optimized `isParentOf` instead in #2756